### ISSUE TITLE
Uploading a exported file into a different directory doesn't work

### ISSUE THREAD

**Issue by doronrosenberg** [maqetta/maqetta#2001]
Created a file in the root and tried to upload it to a subdir and got this error:

"NetworkError: 404 Not Found - http://macwayne2.austin.ibm.com:55556/maqetta/user/doronr@us.ibm.com/ws/workspace/project1/122/lib/dojo/dojo/dojo.js"
dojo.js
require is not defined
/maqetta/()

**Comment by JonFerraiolo** [comment]
Assigned to @waynevicknair as the default migration meister, but first the rest of us can try to figure out what's going on here.

When I tried this, it worked for me. I pasted the file's content into a text editor, saved on my local system, uploaded that file into my project's root folder (as a sibling to Sample1.html) on macwayne2, opened the file, and everything was fine.

What's suspicious is the "122" in the error message. Was there a "122" folder on Maqetta.org? It looks like perhaps the Modify Libraries command had been executed on Maqetta.org to move dojo from project1/lib/dojo to project1/122/lib/dojo.

**Comment by doronrosenberg** [comment]
122 is the directory I created on macwayne2 where I uploaded the file

**Comment by doronrosenberg** [comment]
indeed, uploading into root works fine while uploading into a directory doesn't.

**Comment by jhpedemonte** [comment]
Was this file originally created in a subdir on maqetta.org?

If not, I don't think this is supported and **is not** a migration issue (since the files will stay in the same folders between M4 and M5).

**Comment by JonFerraiolo** [comment]
I don't think there is anything that is feasible for us to do (except document limitations) for the case where users upload HTML files and the HTML files include references to CSS, JS, images, etc, which are in different locations on the new Maqetta workspace than the were on the old Maqetta workspace.

Note that we do munge user HTML files whenever the user moves libraries or does a SaveAs operation into a different folder because in these cases we can guess the user's intent, but for other operations where Maqetta isn't completely controlling the operation, it would be difficult to try to fix up user HTML files.

Perhaps we should change the title on this bug to something about providing URL reference fixup features for cases where HTML files are added to workspaces, and remove the milestone because this is likely to pushed to the back of the bug queue.

**Comment by JonFerraiolo** [comment]
Removed M5 milestone.

**Comment by doronrosenberg** [comment]
The file was originally not created in a subdir

**Comment by JonFerraiolo** [comment]
-> @JonFerraiolo M7 Prio-High

**Comment by JonFerraiolo** [comment]
This is about fixing up HTML files that the OK, I've thought about this one. There are things we can do, but I don't see any of them as feasible for the M7 timeframe, so I'm going to push this to M8/High. Here are two approaches we could take:
- When we open an HTML file, and there are CSS, JS and other things that aren't at the location specified in the file, we could have our client-side code run some fix-up heuristics where we try to figure out which folder the HTML file probably was in and then attempt URL fix ups
- Or we could run the same sort of heuristics on the server by modelling the file on the server, but that approach is not feasible for the near future because we don't have any server-side modelling code.

**Comment by JonFerraiolo** [comment]
-> M8 Prio-Med
